mypy/checkexpr.py

```diff
@@ -1103,7 +1103,11 @@ def infer_arg_types_in_context(
         # Fill in the rest of the argument types.
         for i, t in enumerate(res):
             if not t:
-                res[i] = self.accept(args[i])
+                if arg_kinds[i] == ARG_STAR2:
+                    res[i] = self.accept(args[i], self.chk.named_generic_type('typing.Mapping',
+                        [self.named_type('builtins.str'), AnyType(TypeOfAny.special_form)]))
```

Comment:
Unfortunately the `Any` type here may result in false negatives, since it's used for type inference:

```py
def f(**kwargs: int) -> None: pass

f(**{'x': 'y'})  # No error here after this change
```

More generally, it would be great if we can avoid inferring `Any` types in fully annotated code.

Random idea: Use `Mapping[str, <nothing>]` as the context, potentially only if the kwargs expression is an empty dictionary expression (check for it here). This would certainly be a hack, but at least it's a pretty localized hack.

Reply:
I see... Another idea can be to relax the check here to allow the case where `typ` is `dict[<nothing>, <nothing>]`. At first I didn't like this solution as I thought it was less elegant, but if trying to change the inferred type will result in a hacky code like you said, maybe this one is better?
https://github.com/python/mypy/blob/985a20d87eb3a516ff4457041a77026b4c6bd784/mypy/checkexpr.py#L3933-L3949